src/include/log.h

```diff
@@ -5,6 +5,13 @@
  *
  * See the file LICENSE for redistribution information.
  */
+/*
+ * WT_LSN --
+ *	A log sequence number, representing a position in the transaction log.
+ */
+struct __wt_lsn {
+	uint64_t file_offset;
+};
```

Comment:
@sueloverso, my big question here is why we can't have `WT_LSN` as:

```c
union {
    uint64_t file_offset;
    struct {
        uint32_t file;
        uint32_t offset;
    } x;
};
```

I understand that we need to consider endianness in order to compare LSNs with a single 64-bit comparison (which seems like a big win), but we could address that in this one place with `#ifdef WORDS_BIGENDIAN` and avoid lots of macros and all of the bit-shifting.

Is there some reason why that isn't possible?